src/server/game/Achievements/AchievementMgr.cpp

```diff
@@ -2473,6 +2473,9 @@ bool AchievementGlobalMgr::IsStatisticAchievement(AchievementEntry const* achiev
 
 bool AchievementGlobalMgr::IsAverageCriteria(AchievementCriteriaEntry const* criteria) const
 {
+    if (!sAchievementStore.LookupEntry(criteria->referredAchievement))
+        return false;
+
     if ((sAchievementStore.LookupEntry(criteria->referredAchievement))->flags & ACHIEVEMENT_FLAG_AVERAGE)
         return true;
 
```

Comment:
instead of calling it twice you could do

```cpp
auto referencedAchievement = sAchievementStore.LookupEntry(criteria->referredAchievement);

if (!referencedAchievement)
    return false;
    
if (referencedAchievement->flags & ACHIEVEMENT_FLAG_AVERAGE)
    return true;

Reply:
Corrected, also made use of the variable in the if statement below.